server/src/main/java/org/elasticsearch/transport/SniffConnectionStrategy.java

```diff
@@ -228,126 +228,111 @@ private void collectRemoteNodes(Iterator<Supplier<DiscoveryNode>> seedNodesSuppl
             return;
         }
 
-        if (seedNodesSuppliers.hasNext()) {
-            final Consumer<Exception> onFailure = e -> {
-                if (isRetryableException(e)) {
-                    if (seedNodesSuppliers.hasNext()) {
-                        logger.debug(
-                            () -> format("fetching nodes from external cluster [%s] failed moving to next seed node", clusterAlias),
-                            e
-                        );
-                        collectRemoteNodes(seedNodesSuppliers, listener);
-                        return;
-                    }
-                }
+        assert seedNodesSuppliers.hasNext() : "Must have non empty seed nodes to connect";
```

Comment:
Hmm now that I look more closely, I'm not sure this does always hold. We also get here via the two implementations of `AbstractSniffResponseHandler` which do not look like they guard against running out of nodes properly.

Reply:
Yeah, this is in-fact failed in the last build. Seems we have to keep that condition.